src/evm/protocol/uniswap_v2/state.rs

```diff
@@ -196,54 +197,47 @@ impl ProtocolSim for UniswapV2State {
         let x_prime = U256::from_limbs([limbs[0], limbs[1], limbs[2], limbs[3]]);
 
         if x_prime <= reserve_in {
-            return Ok(BigUint::ZERO);
+            return Ok(Trade::new(BigUint::ZERO, BigUint::ZERO));
         }
+        let amount_in = safe_sub_u256(x_prime, reserve_in)?;
+        let amount_out = cpmm_get_amount_out(
+            amount_in,
+            zero2one,
+            self.reserve0,
+            self.reserve1,
+            UNISWAP_V2_FEE_BPS,
+        )?;
 
-        Ok(u256_to_biguint(safe_sub_u256(x_prime, reserve_in)?))
+        Ok(Trade::new(u256_to_biguint(amount_in), u256_to_biguint(amount_out)))
```

Comment:
you have `x_prime` and the equations, you should be able to compute `y_prime` without doing a new swap on the pool?

Reply:
I was doing it for validation but you're right it's unnecessary